lib/ClangImporter/ImportDecl.cpp

```diff
@@ -467,7 +467,7 @@ void ClangImporter::Implementation::addSynthesizedTypealias(
   auto typealias = new (ctx) TypeAliasDecl(SourceLoc(), SourceLoc(), name,
                                            SourceLoc(), nullptr, nominal);
   typealias->setUnderlyingType(underlyingType);
-  typealias->setAccess(AccessLevel::Public);
+  typealias->setAccess(nominal->getFormalAccess());
```

Comment:
In C++, we could have a private type alias to a public type. We could also have a public type alias to a private type. So the access level of the underlying type and the typealias is independent. Not sure about Swift, and it is possible this is only used in scenarios where the two must match just wanted to double check.

Reply:
Yeah at the moment I'm just doing what Clang tells me the access is. I'm writing test cases to see what would happen if I encounter that scenario and I will adjust the code accordingly (probably to inherit the most restrictive access level).

Reply:
The "access-inversion" test that I just checked in tests as many instances I can think of where private declarations leak through the public interface in C++.

As it turns out, those cases all work fine when imported to Swift, even though they break Swift's invariant about public not exposing private. I'm not actually where the exception for C++ is implemented but I suspect that behavior was implemented because we already had to import public decls exposing private decls, prior to this patch.